packages/esm-patient-forms-app/src/forms/forms-see-all.component.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from 'react';
+import dayjs from 'dayjs';
+import first from 'lodash-es/first';
+import ContentSwitcher from 'carbon-components-react/es/components/ContentSwitcher';
+import Switch from 'carbon-components-react/es/components/Switch';
+import FormView from './form-view-see-all.component';
+import styles from './forms.component.scss';
+import { EmptyState, ErrorState } from '@openmrs/esm-patient-common-lib';
+import { useTranslation } from 'react-i18next';
+import { fetchAllForms, fetchPatientEncounters } from './forms.resource';
+import { filterAvailableAndCompletedForms } from './forms-utils';
+import { Encounter, Form } from '../types';
+import EmptyFormView from './empty-form.component';
+import { getObjectFHIR } from './get-FHIR-object';
+
+enum FormViewState {
+  recommended = 0,
+  completed,
+  all,
+}
+
+const Forms = () => {
+  const query = new URLSearchParams(location.search);
+  const patientUuid = query.get('patientUuid');
+  const [patient, setPatient] = useState<fhir.Patient>();
+  getObjectFHIR(patientUuid).then(setPatient);
```

Comment:
Super fishy - this will be done in each render!

Use a side-effect ...

Reply:
The suggested changes have been made.